src/plugins/markdown.ts

```diff
@@ -5,67 +5,156 @@
                     contents: renderedContent,
                     metadata,
                 });
-                const ref = page.reference;
-                if (pageStore.has(ref)) {
-                    console.warn(`Found duplicate reference "${ref}"; overwriting previous data.`);
-                    console.warn("Rename headings or use metadata `reference` key to disambiguate.");
-                }
-                pageStore.set(ref, page);
-                return page;
             })
             .map((page) => {
-                if (page.contents) {
-                    const newContent = page.contents.reduce((array, content) => {
-                        if (typeof content === "string" || content.tag !== "include") {
-                            return array.concat(content);
-                        }
-                        const pageToInclude = pageStore.get(content.value as string);
-                        if (pageToInclude === undefined) {
-                            throw new Error(`Unknown @include reference '${content.value}' in '${page.reference}'`);
-                        }
-                        return array.concat(pageToInclude.contents!);
-                    }, [] as StringOrTag[]);
-                    page.contents = newContent;
-                }
+                // using `reduce` so we can add one or many entries for each node
+                page.contents = page.contents.reduce((array, content) => {
+                    if (typeof content === "string" || content.tag !== "include") {
+                        return array.concat(content);
+                    }
+                    // inline @include page
+                    try {
+                        const pageToInclude = pageStore.remove(content.value);
+                        return array.concat(pageToInclude.contents);
+                    } catch (ex) {
+                        throw new Error(`Unknown @include reference '${content.value}' in '${page.reference}'`);
+                    }
+                }, [] as StringOrTag[]);
                 return page;
             });
-        const docs = mapToObject(pageStore);
-        return { docs };
+        return pageStore;
+    }
+
+    private buildNavTree(pages: PageMap) {
+        // navPage is used to construct the sidebar menu
+        const roots = createNavigableTree(pages, pages.get(this.options.navPage)).children as IPageNode[];
+        // nav page is not a real docs page so we can remove it from output
+        pages.remove(this.options.navPage);
+        roots.forEach((page) => {
+            if (isPageNode(page)) {
+                page.children.forEach((child) => this.nestChildPage(pages, child, page));
+            }
+        });
+
+        return roots;
+    }
+
+    private nestChildPage(pages: PageMap, child: IPageNode | IHeadingNode, parent: IPageNode) {
+        const originalRef = child.reference;
+
+        // update entry reference to include parent reference
```

Comment:
i think all this should be overridable by something like
`reference: (parent: IPageNode,  child: IPageNode | IHeadingNode) => string`
all this business with separator strings can be contained in the default implementation. This way we can do re-writes of "javascript-api" easily.